pkgs/development/tools/build-managers/xmake/default.nix

```diff
@@ -0,0 +1,30 @@
+{ lib, stdenv, fetchgit
+# optional list of extra waf tools, e.g. `[ "doxygen" "pytest" ]`
+
+}:
+let
+in
+stdenv.mkDerivation rec {
+  pname = "xmake";
+  version = "2.6.7";
+
+  src = fetchgit {
+    url = "https://github.com/xmake-io/xmake";
+    rev = "v${version}";
+    sha256 = "sha256-e6V/RM07T6TCCxdIL47M9P7KiNA6v+mMOcSBZkSp3YQ=";
+  };
+
+  enableParallelBuilding = true;
+
+  installPhase = ''
+    make install prefix=$out
```

Comment:
Can't this use `installFlags = [ "prefix=${placeholder "out"}" ];`?

That way, the installPhase is untouched and preInstall and postInstall still get executed